### PR TITLE
fix: properly wait for rendering

### PR DESCRIPTION
**Pull request by Caele**
The table wasn't properly returning the render promise due to how the load data is setup. Adding a promuse hook fixes this.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -35,14 +36,9 @@ export default function supernova(env) {
...
       const [pageInfo, setPageInfo] = useState({ top: 0, height: 100 });
-      const [tableData, setTableData] = useState();
       const [muiParameters] = useState(muiSetup(__OPIONAL_THEME_DEPS__));
 
-      useEffect(() => {
-        manageData(model, layout, pageInfo).then((d) => {
-          setTableData(d);
-        });
-      }, [layout, pageInfo]);
+      const [tableData] = usePromise(() => manageData(model, layout, pageInfo), [layout, pageInfo]);
```
Comment (haxxmaxx) [inline src/index.js:41]: Don't remember how `usePromise` works. `tableData` will update when `layout` or `pageInfo` changes?

**Review by haxxmaxx — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (Caele) [reply src/index.js:41]: hmnm, didn't acutally test, seems not, I'll check

Comment (Caele) [reply src/index.js:41]: yea it works as expected afaik

Comment (LiKang6688) [reply src/index.js:41]: I think it workss the same as before but more concisely

**Review by LiKang6688 — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (LiKang6688) [reply src/index.js:41]: https://qlik.dev/apis/javascript/nebulajs-stardust#%23%2Fentries%2FusePromise the example here is not the best I would say

**Pull request merged**